cardstack/cardstack
# Fix the back button for the themer

Closes #1199

## Added
- themer logic and layout are copied into a new route

## Changed
- The close button for the themer is now "Save and Close"
- Save and Close button width
- Testing: Replaced monaco editor with a simple `<code>` element for speedier testing/renders
- Testing: renamed the test selector for the card json
- Testing: update existing tests to use the code selector
-  Testing: moved themer tests out of view tests and into editing-css-tests

## Removed 
- logic and layout for the themer from view
- reliance on the editingCss query param throughout the app

The close button is now "Save and Close" because, since the themer route is a sibling of view, the model hook reruns and wipes out the changes.

## How to review:
The diff on this PR is pretty big, but it's mostly copy & paste and tests. I am not sure why some older commits by Will show up on the log for this PR. Possibly I branched from something other than master by mistake.

- try navigating to and from the themer, using in-app links/buttons as well as the browser "back" button
- see that changes made to the CSS get saved successfully, and you can see the preview
- check for any code quality issues
